pkg/cmd/roachtest/operations/changefeeds/utils.go

```diff
@@ -0,0 +1,384 @@
+		}
+		// Check if the job has reached one of the expected final states (e.g., paused or canceled).
+		for _, expectedFinalState := range expectedFinalStates {
+			if jobToAction.state == expectedFinalState {
+				if jobToAction.state == jobs.StateRunning {
+					// we need to wait until the high-water timestamp is set for job state "running"
+					if !jobToAction.highWaterTimestamp.IsEmpty() {
+						return nil
+					}
+				} else {
+					return nil
+				}
+			}
+		}
+		select {
+		case <-ctxTimeout.Done():
+			// If the timeout is exceeded, the operation is aborted
+			return fmt.Errorf("%s on job %s failed due to timeout", action, jobToAction.jobID)
+		case <-time.After(pollForStateInterval):
+			// the current state of the job is obtained.
+			jobToAction.state, jobToAction.highWaterTimestamp, err = getJobIDStateAndHighWM(ctxTimeout, conn, jobToAction.jobID)
+			if err != nil {
+				return err
+			}
+		}
+	}
+}
+
+// checkAndCreateChangefeed checks if there are any existing changefeed jobs,
+// and if the createOnlyIfNoJobs flag is true, it returns early if any jobs exist.
+// If the number of changefeed jobs is less than the maximum allowed limit,
+// it randomly selects a database and table, then attempts to create a new changefeed job for that table.
+// The function returns true if it successfully creates or evaluates the need for a changefeed,
+// and false if any errors occur during the process.
+func checkAndCreateChangefeed(
+	ctx context.Context, o operation.Operation, conn *gosql.DB, createOnlyIfNoJobs bool,
+) error {
+	// Fetch all changefeed jobs that are running or paused.
+	allCFJobs, err := fetchAllChangefeedJobsWithStates(ctx, conn, jobs.StateRunning, jobs.StatePaused)
+	if err != nil {
+		return err
+	}
+	// If there are existing changefeeds and the function is instructed to
+	// create only if no jobs exist, return early.
+	if len(allCFJobs) > 0 && createOnlyIfNoJobs {
+		o.Status(fmt.Sprintf("No jobs created as %d active jobs already exists", len(allCFJobs)))
+		return nil
+	}
+
+	// Get the max allowed changefeeds from environment variables.
+	mcf, err := helpers.EnvOrDefaultInt(maxChangefeeds, defaultEnvValuesInt[maxChangefeeds])
+	if err != nil {
+		return err
+	}
+	if len(allCFJobs) >= mcf {
+		o.Status(fmt.Sprintf("total changefeeds %d are more than or equal to the maximum changefeeds criteria %d",
+			len(allCFJobs), mcf))
+	} else {
+		// If the number of existing changefeeds is below the max limit, create a new one.
+		// Select a random database and table to create the changefeed for.
+		dbName := helpers.PickRandomDB(ctx, o, conn, helpers.SystemDBs)
+		tableName := helpers.PickRandomTable(ctx, o, conn, dbName)
+
+		// Attempt to create the changefeed.
+		err = createChangefeed(ctx, o, conn, allCFJobs, dbName, tableName)
+		if err != nil {
+			return err
+		}
+		// createdJob is the job that was just created. It is identified by querying the
```

Comment:
can't you just use the return value of the `CREATE CHANGEFEED` statement? that's the job id

Reply:
I will be merging this PR and we will take this up when the sink changes are worked on.